src/org/exist/backup/ConsistencyCheck.java

```diff
@@ -324,6 +324,10 @@ public Object start() {
                         EmbeddedXMLStreamReader reader = null;
                         try {
                             final ElementImpl             root            = (ElementImpl)doc.getDocumentElement();
+                            if (root == null) {
+                                return new ErrorReport.ResourceError(ErrorReport.RESOURCE_ACCESS_FAILED, "Failed to access data of document " + doc.getDocumentURI() );
```

Comment:
Looks good in principle, thank you.

I think we could improve the error message though, perhaps something like:

```java
ErrorReport.ResourceError(ErrorReport.NODE_HIERARCHY, "Document has no document element: " + doc.getURI());
```

Reply:
Or "Document has no root element"? (Not sure what a "document element" is?)

Reply:
@joewiz i.e. https://docs.oracle.com/javase/8/docs/api/org/w3c/dom/Document.html#getDocumentElement-- ;-)

Reply:
Sorry, I'm too XDM-centric! ;)